plugins/language_awk.lua

```diff
@@ -0,0 +1,83 @@
+-- mod-version:3
+local syntax = require "core.syntax"
+
+syntax.add {
+  name = "Awk script",
+  files = "%.awk$",
+  headers = "^#!.*bin.*awk",
+  comment = "#",
+  patterns = {
+    -- $# is a awk special variable and the '#' shouldn't be interpreted
+    -- as a comment.
+    { pattern = "$[%a_@*#][%w_]*",                type = "keyword2" },
+    -- Comments
+    { pattern = "#.*",                            type = "comment"  },
+    -- Strings
+    { pattern = { '"', '"', '\\' },               type = "string"   },
+    { pattern = { "'", "'", '\\' },               type = "string"   },
+    { pattern = { '`', '`', '\\' },               type = "string"   },
+    -- Ignore numbers that start with dots or slashes
+    { pattern = "%f[%w_%.%/]%d[%d%.]*%f[^%w_%.]", type = "number"   },
+    -- Operators
+    { pattern = "[!<>|&%[%]:=*]",                 type = "operator" },
+    -- Match parameters
+    { pattern = "%f[%S][%+%-][%w%-_:]+",          type = "function" },
+    { pattern = "%f[%S][%+%-][%w%-_]+%f[=]",      type = "function" },
+    -- Prevent parameters with assignments from been matched as variables
+    {
+      pattern = "%s%-%a[%w_%-]*%s+()%d[%d%.]+",
+      type = { "function", "number" }
```

Comment:
```suggestion
      pattern = "%s%-%a[%w_%-]*()%s+()%d[%d%.]+",
      type = { "function", "normal", "number" }
```

Reply:
What is this matching? Why is there a whitespace match in front?

Reply:
I have no idea, this is from `language_sh`.

Applying the changes doesn't seem to change anything.